### PR TITLE
test: adds unit tests for devices

### PR DESCRIPTION
**Pull request by barneytron**
closes [AB#5499](https://dev.azure.com/rbheBoards/c7d9d701-a85f-45ff-91f6-26c56ff2c4de/_workitems/edit/5499)

## PR Checklist

- [ ] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: src/routes/devices/tags.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+afterEach(() => {
+  jest.clearAllMocks()
+  jest.resetModules()
+})
+
+describe('tags', () => {
+  it('should set status to 200 when getting results from getDistinctTags', async () => {
+    const expectedResult = {}
+    const req = {
```
Comment (rsdmike) [inline src/routes/devices/tags.test.ts:32]: just a suggestion, could put the defintion of req in the beforeEach to reduce code duplication and then just set req.db.devices.getDistinctTags when it needs to change (ie. only in error calls)

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/routes/devices/disconnect.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    guid: '00000000-0000-0000-0000-000000000000'
+  },
+  db: {
+    devices: {
+    }
+  }
+} as any
+
+describe('disconnect', () => {
+  it('should disconnect device if it is connected', async () => {
```
[the remark this answers is not in the document]
Comment (barneytron) [reply src/routes/devices/disconnect.test.ts:42]: Good catch. Added an expectation for destroy to be called in next commit.

File: src/routes/devices/disconnect.test.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      ciraSocket: {
+        destroy: jest.fn()
+      }
+    } as any
+    const guid = req.params.guid
+    await disconnect(req, res as any)
+    expect(jsonSpy).toHaveBeenCalledWith({ success: 200, description: `CIRA connection disconnected : ${guid}` })
+  })
+
+  it('should set status to 404 and not try to disconnect device if it is not connected', async () => {
```
[the remark this answers is not in the document]
Comment (barneytron) [reply src/routes/devices/disconnect.test.ts:53]: Removed the if/else as well as the test in next commit

File: src/routes/devices/tags.test.ts (hunk 1 above)
Comment (barneytron) [reply src/routes/devices/tags.test.ts:32]: refactored to remove duplication in next commit

File: src/routes/devices/disconnect.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (barneytron) [reply src/routes/devices/disconnect.test.ts]: I think removing the if/else caused MPS API issues:

1.  AssertionError  Status code is 404                                                                                       
                     expected response to have status code 404 but got 500                                                    
                     at assertion:0 in test-script                                                                            
                     inside "Devices / Disconnect CIRA"                                                                       
                                                                                                                              
 2.  AssertionError  Device should not be found                                                                               
                     expected 'Internal Server Error' to equal 'Device not found/connected. Please connect again using CIRA.' 
                     at assertion:1 in test-script                                                                            
                     inside "Devices / Disconnect CIRA"

[the remark this answers is not in the document]
Comment (barneytron) [reply src/routes/devices/disconnect.test.ts]: All MPS API tests passing now after adding the ciraMiddleware to disconnect entry in src/routes/devices/index.ts

**Review by rsdmike — APPROVED** [review]

**Pull request merged**